### PR TITLE
fix: corrupt data in routes() response due to healthchecker data

### PR DESCRIPTION
**Pull request by Revolyssup**
### Description

Fixes # (issue)
Currently there is an intermittent bug. Below are the steps to reproduce:
Reproduction steps
1. Create a route and enable health check

```bash
curl http://127.0.0.1:9180/apisix/admin/routes/1 -H 'X-API-KEY: wqGBivhkYbFXiMdcMEAOMZanCGnLSbWE' -X PUT -d '
{
    "uri": "/*",
    "upstream": {
         "nodes": {
            "httpbin.org:80": 1,
            "mockbin.org:80": 1
        },
        "type": "roundrobin",
        "checks": {
            "active": {
                "timeout": 5,
                "type": "tcp",
                "healthy": {
                    "interval": 2,
                    "successes": 1
                },                
                "unhealthy": {
                    "interval": 2,
                    "successes": 1
                }
            }
        }
    }
}'
```

2. **access routes to trigger health checks**

```bash
curl http://127.0.0.1:9080/get
```

3. **When accessing the control api, 500 will be returned and the error in the problem description will be generated.**
### Checklist

This error is intermittent and multiple tries might be needed to reproduce. 
On debugging it was found that the cause of issue was http_router.user_routes table being polluted by adding healthchecks leaving unexpected data which while trying to parse, the error was generated.

As you can see in the generated logs, user_route returned healthchecker data.
Screenshot from 2024-12-19 15-11-00
Screenshot from 2024-12-19 15-10-51

## Fix
Remove healthcheck related objects before sending to user.
- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by membphis** [comment]
we can not update the watched data from etcd, it is dangerous

**Comment by Revolyssup** [comment]
> we can not update the watched data from etcd, it is dangerous

The original data is not being modified. It is already deepcopied. But to make it easier to review I have refactored to move the modification logic just below the deepcopy function so that it's clear that the original data is not being modified.

File: apisix/control/v1.lua [hunk 1]
```diff
@@ -269,6 +269,13 @@ local function iter_add_get_routes_info(values, route_id)
         if new_route.value.upstream and new_route.value.upstream.parent then
             new_route.value.upstream.parent = nil
         end
+        -- remove healthcheck info
+        new_route.checker = nil
```
Comment (membphis) [inline apisix/control/v1.lua:273]: @nic-chen pls take a look when you have time

this changing seem fine to me, but I can not sure 100%

Comment (nic-chen) [reply apisix/control/v1.lua:273]: No problem, I discussed it with @Revolyssup  yesterday

File: apisix/control/v1.lua [hunk 2]
```diff
@@ -269,6 +269,13 @@ local function iter_add_get_routes_info(values, route_id)
...
+        new_route.checker_idx = nil
+        new_route.checker_upstream = nil
+        if new_route.clean_handlers then
+            new_route.clean_handlers = {}
+        end
```
Comment (nic-chen) [inline apisix/control/v1.lua:278]: why not just also set to nil

Comment (Revolyssup) [reply apisix/control/v1.lua:278]: some existing test cases grep logs and expect this to {}

Comment (nic-chen) [reply apisix/control/v1.lua:278]: Then you can fix the test case as well.
Checking the logs has no meaning for the control api.

File: apisix/control/v1.lua [hunk 3]
```diff
@@ -269,6 +269,13 @@ local function iter_add_get_routes_info(values, route_id)
         if new_route.value.upstream and new_route.value.upstream.parent then
             new_route.value.upstream.parent = nil
         end
+        -- remove healthcheck info
```
Comment (nic-6443) [inline apisix/control/v1.lua:272]: we need remove healthcheck info from `iter_add_get_services_info` too

Comment (Revolyssup) [reply apisix/control/v1.lua:272]: done

File: apisix/control/v1.lua (hunk 2 above)
Comment (Revolyssup) [reply apisix/control/v1.lua]: done

**Review by nic-chen — COMMENTED** [review]
please add test cases to ensure error is gone

**Comment by Revolyssup** [comment]
> please add test cases to ensure error is gone

added

**Review by nic-chen — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by nic-6443 — APPROVED** [review]

**Pull request merged**